### DISCUSSION TITLE
Run Weston RDP backend from  xrdp-sesman?

### DISCUSSION THREAD

**Discussion by jpmorrison** [neutrinolabs/xrdp#2314] [Q&A]
I've used the Weston rdp backend to start a linux gui and access it from an RDP client.
It doesn't handle things like multiple sessions so I was wondering if xrdp-sesman could start weston/rdp the way it starts an X or VNC server.

https://raw.githubusercontent.com/technic/docker-weston-rdp/master/entrypoint.sh

**Comment by matt335672** [comment]
There's a lot to be solved to do this I think.

- On most Linux distros using systemd, the way `systemd --user` works means each user can only have one active DISPLAY at any one time.
- Access to port 3389 would be awkward, as Weston would want to use that as well as the xrdp instance you're connecting to in the first place.
- Which virtual channels are handled by Weston and which would be handled by xrdp-chansrv?

It certainly won't work out-of-the-box.

**Reply by jpmorrison** [reply]
I know Microsoft's Weston fork adds socket/vsocket fd to the rdp backend - would that help with the tcp port access?
The RDP backend uses channels for RAIL, clipboard and audio - I don't know how xrdp-chansrv requires.

Maybe a better question is any of this worth it just for Wayland? The graphics in WSLg (Windows services for Linux) are based on Weston RDP backend. Some graphics apps are a bit smoother, perhaps because of GPU being passed through.
